pkg/cli/alpha/internal/update.go

```diff
@@ -0,0 +1,538 @@
+// Kubebuilder scaffolding) into the merge branch (containing user's current state).
+// If conflicts occur, it warns the user to resolve them manually rather than failing.
+func (opts *Update) mergeUpgradeIntoMerge() error {
+	gitCmd := exec.Command("git", "merge", "upgrade")
+	err := gitCmd.Run()
+	if err != nil {
+		// Check if the error is due to merge conflicts (exit code 1)
+		if exitErr, ok := err.(*exec.ExitError); ok && exitErr.ExitCode() == 1 {
+			log.Warn("Merge with conflicts. Please resolve them manually")
+			return nil // Don't treat conflicts as fatal errors
+		}
+		return fmt.Errorf("failed to merge the upgrade branch into the merge branch: %w", err)
+	}
+
+	// Run make targets to ensure all the necessary components are generated,
+	// formatted and linted.
+	log.Info("Running 'make manifests generate fmt vet lint-fix'")
+	if err := opts.runMakeTargets(); err != nil {
+		return fmt.Errorf("failed to run make: %w", err)
+	}
+	log.Info("Successfully ran make targets in merge")
+
+	return nil
+}
+
+// Validate checks if the user is in a git repository and if the repository is in a clean state.
+// It also validates if the version specified by the user is in a valid format and available for
+// download as a binary.
+func (opts *Update) Validate() error {
+	// Validate git repository
+	if err := opts.validateGitRepo(); err != nil {
+		return fmt.Errorf("failed to validate git repository: %w", err)
+	}
+
+	// Validate --from-branch
+	if err := opts.validateFromBranch(); err != nil {
+		return fmt.Errorf("failed to validate --from-branch: %w", err)
+	}
+
+	// Load the PROJECT configuration file
+	projectConfigFile, err := opts.loadConfigFile()
+	if err != nil {
+		return fmt.Errorf("failed to load the PROJECT file: %w", err)
+	}
+
+	// Extract the cliVersion field from the PROJECT file
+	opts.CliVersion = projectConfigFile.Config().GetCliVersion()
+
+	// Determine which Kubebuilder version to use for the update
+	if err := opts.defineFromVersion(); err != nil {
+		return fmt.Errorf("failed to define version: %w", err)
+	}
+
+	// Validate if the specified version is available as a binary in the releases
+	if err := opts.validateBinaryAvailability(); err != nil {
+		return fmt.Errorf("failed to validate binary availability: %w", err)
+	}
+
+	return nil
+}
+
+// Load the PROJECT configuration file to get the current CLI version
+func (opts *Update) loadConfigFile() (store.Store, error) {
+	projectConfigFile := yaml.New(machinery.Filesystem{FS: afero.NewOsFs()})
+	// TODO: assess if DefaultPath could be renamed to a more self-descriptive name
+	if err := projectConfigFile.LoadFrom(yaml.DefaultPath); err != nil {
+		if _, statErr := os.Stat(yaml.DefaultPath); os.IsNotExist(statErr) {
+			return projectConfigFile, fmt.Errorf("no PROJECT file found. Make sure you're in the project root directory")
+		}
+		return projectConfigFile, fmt.Errorf("fail to load the PROJECT file: %w", err)
+	}
+	return projectConfigFile, nil
```

Comment:
We should use the same implementation that's already used by alpha generate to avoid duplication and prevent any unexpected differences in behavior.